src/automation/tasks/scripts/guidellm_script.py

```diff
@@ -39,36 +50,76 @@ def main(configurations=None):
     if isinstance(force_download, str):
         force_download = force_download.lower() == "true"
 
-
     # Resolve model_id
     model_id = resolve_model_id(args["Args"]["model"], clearml_model, force_download)
 
+    gpu_count = int(guidellm_args.get("gpu_count", 1)) 
+
     # Start vLLM server
     server_process, server_initialized, server_log = start_vllm_server(
         vllm_args,
         model_id,
         guidellm_args["target"],
         args["Args"]["server_wait_time"],
+        gpu_count,
     )
 
     if not server_initialized:
         kill_process_tree(server_process.pid)
         task.upload_artifact(name="vLLM server log", artifact_object=server_log)
-        raise AssertionError("Server failed to intialize")
+        raise AssertionError("Server failed to initialize")
 
     # Parse through environment variables
     for k, v in environment_args.items():
         os.environ[k] = str(v)
 
     guidellm_args["model"] = model_id
 
-    from guidellm import generate_benchmark_report
-    guidellm_args = cast_args(guidellm_args, generate_benchmark_report)
-    report = generate_benchmark_report(**guidellm_args)
-    kill_process_tree(server_process.pid)
+    import json
+    import asyncio
+    from pathlib import Path
+    from guidellm.benchmark.entrypoints import benchmark_with_scenario
+    from guidellm.benchmark.scenario import GenerativeTextScenario, get_builtin_scenarios
+
+    user_scenario = guidellm_args.get("scenario", "")
+    if user_scenario:
```

Comment:
Let's add a comment here to clarify that all these scenarios are to be defined in guidellm, not here. This is a temporary solution